File: 0860/solution.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        int[] cashier = { 0, 0 };
        for (int bill : bills) {
            // pay with a 5
            if (bill == 5) {
                cashier[0]++;
                // pay with a 10
            } else if (bill == 10) {
                cashier[1]++;
                if (cashier[0] == 0) {
                    return false;
                }
                cashier[0]--;
                // pay with a 20
            } else {
                if (cashier[1] > 0 && cashier[0] > 0) {
                    cashier[1]--;
                    cashier[0]--;
                } else if (cashier[0] >= 3) {
                    cashier[0] -= 3;
                } else {
                    return false;
                }
            }
        }
        return true;
    }
}

// time O(n)
// space O(1)

File: 0860/solution.py
class Solution:
    def lemonadeChange(self, bills: list[int]) -> bool:
        cashier = [0, 0]
        for bill in bills:
            if bill == 5:
                cashier[0] += 1
            elif bill == 10:
                cashier[1] += 1
                if cashier[0] == 0:
                    return False
                cashier[0] -= 1
            else:
                if cashier[1] > 0 and cashier[0] > 0:
                    cashier[1] -= 1
                    cashier[0] -= 1
                elif cashier[0] >= 3:
                    cashier[0] -= 3
                else:
                    return False
        return True
